xdsl/dialects/snitch_runtime.py

```diff
@@ -235,6 +235,231 @@ class DmaWaitAllOp(SnitchRuntimeBarrier):
     name = "snrt.dma_wait_all"
 
 
+"""
+The different SSR data movers.
+    SNRT_SSR_DM0 = 0,
+    SNRT_SSR_DM1 = 1,
+    SNRT_SSR_DM2 = 2,
+"""
+ssr_dm = i32
+
+"""
+The different dimensions.
+    SNRT_SSR_1D = 0,
+    SNRT_SSR_2D = 1,
+    SNRT_SSR_3D = 2,
+    SNRT_SSR_4D = 3,
+"""
+ssr_dim = i32
+
+
+@irdl_op_definition
+class SsrLoop1dOp(SnitchRuntimeBaseOp, ABC):
+    """
+    Configure an SSR data mover for a 1D loop nest.
+    """
+
+    name = "snrt.ssr_loop_1d"
+    dm: Operand = operand_def(ssr_dm)
+    b0: Operand = operand_def(IndexType)
+    i0: Operand = operand_def(IndexType)
```

Comment:
```suggestion
    indices: VarOperand = var_operand_def(IndexType)
```

Reply:
Then in a `verify_` function please add a check that bounds and indices are the same length